wiki/DEFINITION_OF_DONE_CRITERIA.md

```diff
@@ -0,0 +1,99 @@
+# Defining cuML's Definition of Done Criteria
+
+
+## Algorithm Completion Checklist
+
+Below is a quick and simple checklist for developers to determine whether an algorithm is complete and ready for release. Most of these items contain more detailed descriptions in their corresponding developer guide. The checklist is broken down by layer (C++ or Python) and categorized further into
+
+- **Design:** All algorithms should be designed with an eye on maintainability, performance, readability, and robustness.
+- **Testing:** The goal for automated testing is to increase the code coverage as much as possible in order to ease time spent fixing bugs and developing new features. Additionally, a very important factor for a tool like `cuml` is to provide testing with multitple datasets that really stress the mathematical behavior of the algorithms, even if they don't increase code coverage! A comprehensive set of tests lowers the possibility for regressions and the introduction of bugs as the code evolves between versions. This covers both correctness & performance. 
```

Comment:
Not quite sure I got the message here. Why add and run any more extra tests if they do not improve coverage? May be we should reword it to "add multiple datasets precisely to help improve coverage"?

Reply:
This was a change requested by @dantegd and I think he's defining coverage as a lateral, physical path through the code, rather than better covering the space of possible inputs (not necessarily taking different physical paths through the code). 

Semantics aside, I do agree with you, @teju85, that the wording `even if they don't improve code coverage` can, and should, be improved.

Reply:
@teju85 @dantegd I've adjusted the wording for this bullet point based on our discussion. Can you guys check the updated version and let me know if this works for you?